tests/onnx/test_onnx_v2.py

```diff
@@ -94,6 +133,67 @@ def test_flatten_output_collection_property(self):
             },
         )
 
+    def test_determine_framework(self):
```

Comment:
I suggest promoting this to a standalone class like `DetermineFrameWorkTest` and then treat each case with a dedicated function like `test_framework_provided(self)` etc

Reply:
Thanks for the suggestion! Done in [67416f2](https://github.com/huggingface/transformers/pull/18615/commits/67416f22094bb29b7bdcfe22854ee00ef4e1e112). I broke it down to the 3 main paths, hope that is sufficient.